crates/sdl-encoder/src/field_value.rs

```diff
@@ -0,0 +1,120 @@
+use std::fmt::{self, Display};
+
+/// Convenience Field Value implementation used when creating a Field.
+/// Can be a `Type`, a `NonNull` or a `List`.
+///
+/// This enum is resposible for encoding creating values such as `String!`, `[[[[String]!]!]!]!`, etc.
+///
+/// ### Example
+/// ```rust
+/// use sdl_encoder::{FieldValue};
+///
+/// let field_ty = FieldValue::Type {
+///     ty: "String".to_string(),
+/// };
+///
+/// let list = FieldValue::List {
+///     ty: Box::new(field_ty),
+/// };
+///
+/// let non_null = FieldValue::NonNull { ty: Box::new(list) };
+///
+/// assert_eq!(non_null.to_string(), "[String]!");
+/// ```
+#[derive(Debug, PartialEq, Clone)]
+pub enum FieldValue {
```

Comment:
Where does the name `FieldValue` come from? Isn't this a representation of any GraphQL type, that would also be used for the type of field and directive arguments? Maybe call this `Type` or `GraphQLType`, and refer to the `Type` variant as `NamedType`?

Reply:
I find it confusing to call it a `GraphQLType`, even more so when it's just `Type`. There isn't anything in the spec about this particular bit of SDL, so I figured we'd call this something that's a bit more easily understood. 

`NamedType` sounds good! I'll use that instead.

Reply:
It is called [`Type`](http://spec.graphql.org/draft/#Type) in the spec, and that also lists `NamedType` as part of the recursive definition. Note that `NamedType` is not a replacement for `FieldValue` however, it is just one of the possible values of the enum (besides `NonNull` and `List`).

Reply:
If you don't like `Type`, maybe `TypeRef` would be an alternative?

Reply:
Ok, we will have it as `Type_`, since `Type` is reserved.